### PR TITLE
Adding test to verify image update for WLS domain 

### PR DESCRIPTION
**Pull request by maggiehe00**
Adding test to verify image update for WLS domain.

Kind new parallel test result 9/30
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2195/testReport/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 1]
```diff
@@ -848,6 +858,83 @@ public void testCreateNewCluster() {
 
   }
 
+  /**
+   * Modify the domain scope property
+   * From: "image: container-registry.oracle.com/middleware/weblogic:12.2.1.4" to
+   * To: "image: container-registry.oracle.com/middleware/weblogic:14.1.1.0-11"
+   * Verify all the pods are restarted and back to ready state
+   * Verify the admin server is accessible and cluster member is healthy
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:866]:
```suggestion
   * Verify the admin server is accessible and cluster members are healthy
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 2]
```diff
@@ -848,6 +858,83 @@ public void testCreateNewCluster() {
...
+    for (int i = 1; i <= replicaCount; i++) {
+      managedServerNames.add(managedServerNameBase + i);
+    }
+
+    // get the original domain resource before update
+    Domain domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    assertNotNull(domain1, "Got null domain resource");
+    assertNotNull(domain1.getSpec(), domain1 + "/spec is null");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:893]:
```suggestion
    assertNotNull(domain1.getSpec(), domain1 + " /spec is null");
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 3]
```diff
@@ -848,6 +858,83 @@ public void testCreateNewCluster() {
...
+   * Modify the domain scope property
+   * From: "image: container-registry.oracle.com/middleware/weblogic:12.2.1.4" to
+   * To: "image: container-registry.oracle.com/middleware/weblogic:14.1.1.0-11"
+   * Verify all the pods are restarted and back to ready state
+   * Verify the admin server is accessible and cluster member is healthy
+   */
+  @Order(5)
+  @Test
+  @DisplayName("Verify server pods are restarted by updating image name")
+  @Slow
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:871]: Remove this tag @Slow. It is not being used

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 4]
```diff
@@ -848,6 +858,83 @@ public void testCreateNewCluster() {
...
+    assertNotNull(domain1, "Got null domain resource");
+    assertNotNull(domain1.getSpec(), domain1 + "/spec is null");
+
+    // get the map with server pods and their original creation timestamps
+    podsWithTimeStamps = getPodsWithTimeStamps(domainNamespace, adminServerPodName, managedServerPodNamePrefix,
+        replicaCount);
+
+    //print out the original image name
+    String imageName = domain1.getSpec().getImage();
+    logger.info("Original domain image name is: {0}", imageName);
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:901]:
```suggestion
    logger.info("Currently used domain image name is: {0}", imageName);
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 5]
```diff
@@ -848,6 +858,83 @@ public void testCreateNewCluster() {
...
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/image\",")
+        .append("\"value\": \"")
+        .append(imageUpdate)
+        .append("\"}]");
+    logger.info("PatchStr for imageUpdate: {0}", patchStr.toString());
+
+    boolean cmPatched = patchDomainResource(domainUid, domainNamespace, patchStr);
+    assertTrue(cmPatched, "patchDomainCustomResource(imageUpdate) failed");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:914]:
```suggestion
    assertTrue(patchDomainResource(domainUid, domainNamespace, patchStr), "patchDomainCustomResource(imageUpdate) failed");
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 6]
```diff
@@ -848,6 +858,83 @@ public void testCreateNewCluster() {
...
+
+    domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    assertNotNull(domain1, "Got null domain resource after patching");
+    assertNotNull(domain1.getSpec(), domain1 + "/spec is null");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:920]:
```suggestion
    assertNotNull(domain1.getSpec(), domain1 + " /spec is null");
```

**Review by sankarpn — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java [hunk 7]
```diff
@@ -944,6 +1031,26 @@ private static void verifyMemberHealth(String adminServerPodName, List<String> m
         });
   }
 
+  private Map getPodsWithTimeStamps(String domainNamespace, String adminServerPodName,
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java:1034]: Can it be moved to utility class. This is being used in many tests when we verify a domain restart.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java (hunk 1 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java (hunk 2 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java (hunk 4 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java (hunk 5 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java (hunk 6 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java (hunk 3 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java (hunk 7 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java]: Done

**Review by sankarpn — APPROVED** [review]
LGTM

**Comment by anpanigr** [comment]
LGTM

**Review by anpanigr — APPROVED** [review]
LGTM

**Pull request merged**